src/rosdiscover/interpreter/summary.py

```diff
@@ -142,3 +207,16 @@ def unresolved(self) -> Iterator[NodeSummary]:
         for n in self._node_to_summary.values():
             if n.placeholder:
                 yield n
+
+    @classmethod
+    def merge(cls, sum1: 'SystemSummary', sum2: 'SystemSummary') -> 'SystemSummary':
+        node_summaries = {}
```

Comment:
type annotation is probably needed here? If not, `t.Dict[t.Any]` will be assumed, I think? (Could be wrong.)